plugin-modernizer-core/src/main/resources/META-INF/rewrite/recipes.yml

```diff
@@ -181,3 +181,12 @@ recipeList:
       filePattern: updatecli/values.yml
   - org.openrewrite.DeleteSourceFiles:
       filePattern: updatecli/**/*.yml
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: io.jenkins.tools.pluginmodernizer.AddOrModernizeJenkinsFile
+displayName: Add Jenkinsfile
+description: Updates Jenkinsfile to build with recommended Java versions, platforms, and settings, or creates the file if absent. Also upgrades to the next parent major version, as the new version of the Jenkinsfille uses JDK17 and JDK21.
+tags: ['chore']
+recipeList:
+  - org.openrewrite.jenkins.ModernizeJenkinsfile
+  - io.jenkins.tools.pluginmodernizer.UpgradeNextMajorParentVersion
```

Comment:
This will move to parent 5.x ? Is it expected ?

I think it perhaps a bit premature. Shouldn't we use `io.jenkins.tools.pluginmodernizer.UpgradeParentVersion` instead ?

Reply:
You're right, I went full throttle and maybe missed the initial target.
Let me try with `io.jenkins.tools.pluginmodernizer.UpgradeParentVersion`and see where this is going.

Reply:
It's way more reasonable this way.
It produced: https://github.com/jenkinsci/vagrant-plugin/pull/13/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R12.